src/lib.rs

```diff
@@ -228,6 +255,20 @@ pub enum OsMode {
     Interrupt,
 }
 
+#[derive(Debug, Clone, Copy)]
+/// Device Resolution
+pub enum Resolution {
```

Comment:
```suggestion
pub (crate) enum Resolution {
```
This does not need to be public, right?